plugins/modules/slack.py

```diff
@@ -297,10 +297,7 @@ def build_payload_for_slack(text, channel, thread_id, username, icon_url, icon_e
         # With a custom color we have to set the message as attachment, and explicitly turn markdown parsing on for it.
         payload = dict(attachments=[dict(text=escape_quotes(text), color=color, mrkdwn_in=["text"])])
     if channel is not None:
-        if channel.startswith(('#', '@', 'C0', 'GF', 'G0', 'CP')):
-            payload['channel'] = channel
-        else:
-            payload['channel'] = '#' + channel
+        payload['channel'] = channel
```

Comment:
While I like this, this creates a new problem: it breaks backwards compatibility, and will make the module stop working for folks who have been using it successfully so far.

The best way to do this is to add a new option which controls this behavior (with default being the current behavior). Then in the next major release we can deprecate the current default, and in the major release afterwards flip the default to the new behavior. This will allow users to start using the new behavior now, inform them about the switch soon, and give them some time to adjust, or simply to explicitly chose the behavior they prefer.

Maybe a three-value option would be best: `prepend_hash: always | never | auto`, with current default `auto` (we can also not switch to a new default, but simply make this required in the future). What do you think?